src/apps/pillarx-app/components/HorizontalToken/HorizontalToken.tsx

```diff
@@ -0,0 +1,38 @@
+// images
+import defaultLogo from '../../images/logo-unknown.png';
+
+// components
+import BodySmall from '../Typography/BodySmall';
+import Body from '../Typography/Body';
+import TokensPercentage from '../TokensPercentage/TokensPercentage';
+
+type HorizontalTokenProps = {
+    tokenIndex: number;
+    logo?: string;
+    tokenSymbol?: string;
+    tokenName?: string;
+    tokenValue?: number;
+    percentage?: number;
+    isLast?: boolean;
+};
+
+const HorizontalToken = ({ tokenIndex, logo, tokenSymbol, tokenName, tokenValue, percentage, isLast }: HorizontalTokenProps) => {
+    return (
+        <div className={`flex w-full justify-between py-5 ${!isLast && 'border-b border-[#1F1D23] mobile:border-[#27262F]'}`}>
+            <div className='flex items-center'>
+                <BodySmall className='text-purple_light mr-[18px]'>0{tokenIndex}</BodySmall>
+                <img src={logo ?? defaultLogo} className='w-[50px] h-[50px] object-fill rounded mr-3.5' />
+                <div className='flex flex-col'>
+                    {tokenSymbol && <Body>{tokenSymbol}</Body>}
+                    {tokenName && <BodySmall className='text-purple_light'>{tokenName}</BodySmall>}
+                </div>
+            </div>
+            <div className='flex flex-col items-center'>
+                {tokenValue && <Body>${tokenValue.toFixed(4)}</Body>}
+                <TokensPercentage percentage={percentage} />
```

Comment:
Can i just check that this doesn't render if there's no `percentage`?

Reply:
Yes correct, if no percentage the whole component TokensPercentage does not render, the logic is within that component.